packages/react-core/src/hooks/useIsSignedIn.ts

```diff
@@ -0,0 +1,45 @@
+import { Hub } from 'aws-amplify/utils';
+import { useEffect, useState } from 'react';
+import { HubCallback } from '@aws-amplify/core';
+import { getCurrentUser } from 'aws-amplify/auth';
+
+/**
+ * Returns `false`:
+ * - if user has no credentials cached in browser storage
+ * - if user signs out of account
+ *
+ * Returns `true`:
+ * - if user has cached credentials
+ * - if user signs in to account
+ */
+export default function useIsSignedIn(): boolean {
+  const [isSigned, setIsSigned] = useState<boolean>(false);
+
+  useEffect(() => {
+    getCurrentUser()
+      .then(() => {
+        setIsSigned(true);
+      })
+      .catch(() => {
+        setIsSigned(false);
+      });
+  }, []);
+
+  useEffect(() => {
+    const listener: HubCallback = ({ payload }) => {
+      if (payload.event === 'signedIn') {
+        setIsSigned(true);
+      } else if (payload.event === 'signedOut') {
+        setIsSigned(false);
+      }
```

Comment:
Does this work with other auth hub events? I recall there being events like `tokenRefresh_failure `, does `signedOut` get called after those events and does the hook successfully show signed in status for something like that?

Reply:
I think yes, I have to catch that event as well. 

If you look at [useAuth](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/src/hooks/useAuth.ts), it seems like tokenRefresh_failure and signIn_failure have the same functionality, which is different from signedOut and doesn't update the useAuth user state. 

But if you look at [defaultHubHandler](https://github.com/aws-amplify/amplify-ui/blob/main/packages/ui/src/helpers/authenticator/defaultAuthHubHandler.ts), both tokenRefresh_failure and signedOut send out the same event that triggers a signOutActor in the state machine. But I did not find any references to this defaultHubHandler anywhere else in the codebase. 

Also, I asked someone on JS and they said they do not send a signedOut event along with the tokenRefresh_failure event.

Reply:
Sounds good to me